source/common/upstream/cluster_manager_impl.cc

```diff
@@ -715,8 +718,29 @@ Host::CreateConnectionData ClusterManagerImpl::tcpConnForCluster(const std::stri
 
   HostConstSharedPtr logical_host = entry->second->lb_->chooseHost(context);
   if (logical_host) {
-    auto conn_info =
-        logical_host->createConnection(cluster_manager.thread_local_dispatcher_, nullptr);
+
+    absl::optional<std::string> overrideServerName;
+
+    absl::optional<std::string> requestedServerName;
+    if (context->downstreamConnection() &&
+        context->downstreamConnection()
+            ->streamInfo()
+            .filterState()
+            .hasData<OriginalRequestedServerName>(OriginalRequestedServerName::Key)) {
+      const OriginalRequestedServerName& server_name =
+          context->downstreamConnection()
+              ->streamInfo()
+              .filterState()
+              .getData<OriginalRequestedServerName>(OriginalRequestedServerName::Key);
+      requestedServerName = server_name.value();
+    }
+
+    if (entry->second->cluster_info_->forwardOriginalServerNameIndication() && context &&
```

Comment:
You no longer need this? If you don't put the original_sni filter, then you won't hit this.